jetty-ee9/jetty-ee9-websocket/jetty-ee9-websocket-jakarta-server/src/main/java/org/eclipse/jetty/ee9/websocket/jakarta/server/JakartaWebSocketServerContainer.java

```diff
@@ -323,9 +323,9 @@ public void upgradeHttpToWebSocket(Object httpServletRequest, Object httpServlet
             baseRequest.setAttribute(WebSocketConstants.WEBSOCKET_WRAPPED_RESPONSE_ATTRIBUTE, response);
 
             if (handshaker.upgradeRequest(negotiator, baseRequest, baseResponse, callback, components, defaultCustomizer))
-            {
                 callback.block();
-            }
+            else
+                throw new IllegalStateException("Invalid WebSocket Upgrade Request");
```

Comment:
Use `WebSocketException` here.

Reply:
That would be a violation of the contract defined in jakarta websocket 2.1